packages/contracts/contracts/plugin/psp/PluginSetupProcessor.sol

```diff
@@ -0,0 +1,703 @@
+    {
+        bytes32 pluginId = _getPluginId(_dao, _params.setupPayload.plugin);
+
+        PluginInformation storage pluginInformation = states[pluginId];
+
+        bytes32 setupId = _getSetupId(
+            _params.pluginSetupRef,
+            bytes32(0),
+            hHash(_params.setupPayload.currentHelpers),
+            bytes32(0),
+            bytes("")
+        );
+
+        if (pluginInformation.currentSetupId != setupId) {
+            revert InvalidSetupId({
+                currentSetupId: pluginInformation.currentSetupId,
+                setupId: setupId
+            });
+        }
+
+        PluginRepo.Version memory version = _params.pluginSetupRef.pluginSetupRepo.getVersion(
+            _params.pluginSetupRef.versionTag
+        );
+
+        (permissions, actions) = PluginSetup(version.pluginSetup).prepareUninstallation(
+            _dao,
+            _params.setupPayload
+        );
+
+        bytes32 newSetupId = _getSetupId(
+            _params.pluginSetupRef,
+            pHash(permissions),
+            bytes32(0),
+            aHash(actions),
+            bytes("")
+        );
+
+        // Only allow to prepare if setupId has not been prepared before.
+        // Note that the following check ensures that the same setupId can be prepared
+        // once again if the plugin was uninstalled and then installed/updated..
+        if (pluginInformation.blockNumber < pluginInformation.setupIds[newSetupId]) {
+            revert SetupAlreadyPrepared(setupId);
+        }
+
+        pluginInformation.setupIds[newSetupId] = block.number;
+
+        emit UninstallationPrepared({
+            sender: msg.sender,
+            dao: _dao,
+            setupId: newSetupId,
+            pluginSetupRepo: _params.pluginSetupRef.pluginSetupRepo,
+            versionTag: _params.pluginSetupRef.versionTag,
+            setupPayload: _params.setupPayload,
+            permissions: permissions,
+            actions: actions
+        });
+    }
+
+    /// @notice Applies the permissions of a prepared uninstallation to a DAO.
+    /// @param _dao The address of the DAO.
+    /// @param _dao The address of the installing DAO.
+    /// @param _params The struct containing the parameters for the `applyUninstallation` function.
+    /// @dev The list of `_currentHelpers` has to be specified in the same order as they were returned from previous setups preparation steps (the latest `prepareInstallation` or `prepareUpdate` step that has happend) on which the uninstallation was prepared for.
+    function applyUninstallation(address _dao, ApplyUninstall calldata _params)
+        external
+        canApply(_dao, APPLY_UNINSTALLATION_PERMISSION_ID)
+    {
+        bytes32 pluginId = _getPluginId(_dao, _params.plugin);
+
+        PluginInformation storage pluginInformation = states[pluginId];
+
+        bytes32 setupId = _getSetupId(
+            _params.pluginSetupRef,
+            pHash(_params.permissions),
+            _params.helpersHash,
```

Comment:
no need to pass the helpers hash. it is set to `bytes(0)` on line 564

Reply:
fixed in https://github.com/aragon/core/pull/164/commits/30844f8bc5a5291e367ee6167f6182fa456d9cf5